gsa/src/web/components/icon/svgicon.js

```diff
@@ -41,41 +41,78 @@ const Styled = styled.span`
     }
   }
 
+  & svg {
+    background: ${props =>
+      props.loading
+        ? 'url(/img/loading.gif) center center no-repeat'
+        : undefined};
+  }
+
   & svg path {
-    fill: ${({active = true}) => (active ? undefined : Theme.inputBorderGray)};
+    display: ${props => (props.loading ? 'none' : undefined)};
+    fill: ${props =>
+      props.active || props.loading ? undefined : Theme.inputBorderGray};
   }
 `;
 
 const SvgIcon = ({
   disabled = false,
   active = !disabled,
   children,
+  loadingTitle = 'Loading...',
+  title,
   to,
   value,
   onClick,
   ...other
-}) => (
-  <Styled
-    {...other}
-    data-testid="svg-icon"
-    active={active}
-    onClick={
-      isDefined(onClick) && !disabled
-        ? event => {
-            event.preventDefault();
-            event.stopPropagation();
-            onClick(value);
-          }
-        : undefined
+}) => {
+  const [loading, setLoading] = useState(false);
+
+  const handleClick = () => {
+    if (onClick && !loading) {
+      const promise = onClick(value);
+
+      if (isDefined(promise)) {
+        setLoading(true);
+        // eslint-disable-next-line no-shadow
+        promise
+          .then(() => {
+            setLoading(false);
+          })
+          .catch(() => {
+            setLoading(false);
+          });
+      }
     }
-  >
-    {isDefined(to) ? <Anchor href={to}>{children}</Anchor> : children}
-  </Styled>
-);
+  };
+
+  return (
+    <Styled
+      {...other}
+      data-testid="svg-icon"
+      active={active && !loading}
+      loading={loading}
+      title={loading ? loadingTitle : title}
+      onClick={
+        isDefined(onClick) && !disabled && !loading
+          ? event => {
```

Comment:
This code should be moved to he handleClick function. Currently it creates a function that calls another function. We only need one function ;-)